File: server/src/main/java/br/app/pregsrateio/common/erros/ControleFluxoException.java
package br.app.pregsrateio.common.erros;

import lombok.Data;
import lombok.EqualsAndHashCode;
import org.springframework.http.HttpStatus;

import java.util.Collections;
import java.util.List;
import java.util.Map;

import static org.springframework.http.HttpStatus.UNPROCESSABLE_ENTITY;

@Data
@EqualsAndHashCode(callSuper = true)
public class ControleFluxoException extends RuntimeException {
    private final String codigo;
    private final String mensagem;
    private final List<Map<String, Object>> detalhes;
    private final HttpStatus status;

    public ControleFluxoException(String codigo, String mensagem) {
        this(codigo, mensagem, Collections.emptyList(), UNPROCESSABLE_ENTITY);
    }

    public ControleFluxoException(String codigo, String mensagem, HttpStatus status) {
        this(codigo, mensagem, Collections.emptyList(), status);
    }

    public ControleFluxoException(String codigo, String mensagem, List<Map<String, Object>> detalhes, HttpStatus status) {
        super("Erro de controle de fluxo: " + codigo, null, true, false);
        this.codigo = codigo;
        this.mensagem = mensagem;
        this.detalhes = detalhes != null ? detalhes : Collections.emptyList();
        this.status = status != null ? status : UNPROCESSABLE_ENTITY;
    }
}

File: server-old/src/main/kotlin/br/dev/brunorsch/pregsrateio/common/erros/ControleFluxoException.kt
package br.dev.brunorsch.pregsrateio.common.erros

import io.micronaut.http.HttpStatus


class ControleFluxoException(
    val codigo: String,
    val mensagem: String,
    val detalhes: List<Map<String, Any>> = emptyList(),
    val status: HttpStatus = HttpStatus.UNPROCESSABLE_ENTITY,
    override val message: String = "Erro de controle de fluxo: $codigo",
) : RuntimeException(
    message, null, true, false
)